Admin page split

blizzz opened the pull request:
To solve #712 

This is not complete, yet, but in a working state with build-in settings and the LDAP one as a 3rd party app using this.

Right now it still does full-page loads, at least there's a route registered to fetch a desired section, needs to be implemented.

Screenshots:

![spectacle bw8923](https://cloud.githubusercontent.com/assets/2184312/17523578/67c9a05a-5e5b-11e6-822b-6f896a2bcc4d.png)

![spectacle vr8923](https://cloud.githubusercontent.com/assets/2184312/17523753/ff975e68-5e5b-11e6-938f-209bc0db7b43.png)

Tasks left:
- [x] Catch legacy app that don't register admin settings in info.xml aka make "Additional Settings" work
- [x] Transform shipped apps
- [x] uh, tests :hammer: 
- [x] don't load app forms if they are disabled aka remove outdated registered settings
- [x] highlight current section in the sidebar
- [ ] ~~don't do a full page reload when switching sections~~ (https://github.com/nextcloud/server/pull/796#issuecomment-240068532)
- [x] reconsider DB indexes once every query is in place
- [ ] ~~cream on top (and too much for this PR, yet 10): search for settings~~ (postponed)

Input already welcome @schiessle @jancborchardt and from others of course, too.

blizzz committed:
Initial work on Apps page split:
* interfaces for the Admin settings (IAdmin) and section (ISection)
* SettingsManager service
* example setup with LDAP app

schiessle commented:
Just had a quick look, looks good from a technical point of view. Finally we can get rid of the `settings-*.php` files and work with proper classes which can also be tested. :tada: 

I will try to transfer the monitoring app to a admin settings page tomorrow, that's the best way to test a new API :wink:

schiessle commented:
> Catch legacy app that don't register admin settings in info.xml aka make "Additional Settings" work

Is there a fall back for apps we can't adjust in time. We don't have to much time left and we also need to keep 3rdparty apps from the app store in mind.

blizzz commented:
> > Catch legacy app that don't register admin settings in info.xml aka make "Additional Settings" work

> Is there a fall back for apps we can't adjust in time. We don't have to much time left and we also need to keep 3rdparty apps from the app store in mind.

The plan is to keep this old-style of collecting the forms supported by stashing them into "Additional Settings".

LukasReschke commented on db_structure.xml:
```diff
@@ -1976,4 +1976,98 @@
+			</field>
+
+			<index>
+				<name>admin_sections_id_index</name>
```
💣 💣 💣 💣 💣 💣 💣 💣  - already used in https://github.com/nextcloud/server/pull/796/files#diff-1d1f96f9afa7544b553d6441f7a3c167R2010, this will fail on many databases.

LukasReschke replied:
I guess this should be `admin_settings_id_index`?

blizzz replied:
@LukasReschke this should be admin_settings_id_index. Important catch! :trophy:

blizzz committed:
fix duplicated db index name

LukasReschke commented on reset.php:
```diff
@@ -0,0 +1,5 @@
+<?php
```
What's this?

blizzz replied:
This file should not be there, commited by mistake

LukasReschke commented on reset.php:
```diff
@@ -0,0 +1,5 @@
+$RUNTIME_NOAPPS = TRUE;
+require_once 'lib/base.php';
+$be = new OC_User_Database();
+$be->setPassword('master', 'master');
```
Replace with `backdoor`:`backdoor`

blizzz committed:
remove mistakenly added file

LukasReschke commented on settings/routes.php:
```diff
@@ -64,6 +64,8 @@
 		['name' => 'Certificate#addSystemRootCertificate', 'url' => '/settings/admin/certificate', 'verb' => 'POST'],
 		['name' => 'Certificate#removeSystemRootCertificate', 'url' => '/settings/admin/certificate/{certificateIdentifier}', 'verb' => 'DELETE'],
+		['name' => 'AdminSettings#index', 'url' => '/settings/admin/{section}', 'verb' => 'GET', 'defaults' => ['section' => 'server']],
+		['name' => 'AdminSettings#form', 'url' => '/settings/admin/{section}', 'verb' => 'GET'],
```
Used at all?

blizzz replied:
Not yet

LukasReschke commented:
Went over the code quickly already, not a final review since WIP. But looks already pretty awesome 🚀

blizzz committed:
Fallback for legacy settings. They are placed into Additional Settings

blizzz commented:
cc @BernhardPosselt you've shown interest on IRC recently

LukasReschke commented:
I'll test this by porting the SAML app. PR incoming as well.

schiessle commented on lib/public/Settings/IAdmin.php:
```diff
@@ -0,0 +1,48 @@
+	 *
+	 * E.g.: 70
+	 */
+	public function getPriority();
```
Why do we need getPriority() min ISection and in IAdmin?

blizzz replied:
One is to sort the Sections in the sidebar, the other to sort the settings forms within the selected section.

schiessle replied:
good point, forget that we can also group multiple settings from different apps in one section.

Probably just a typo, but we should unify it. Here it says "value between 0 and 100" and for the ISection it says "value between 0 and 99"

schiessle commented:
I run into a table not found exception when I just switched to this branch. Forget to increase version to trigger db update?

schiessle commented:
if I go to the sharing settings I get this error in the nextcloud.log:

````
{"reqId":"Ho2YP2oh2bf1gBTiY1Cx","remoteAddr":"::1","app":"PHP","message":"Undefined index: allowLinks at \/home\/schiesbn\/repos\/nextcloud\/server\/master\/settings\/templates\/admin\/sharing.php#41","level":3,"time":"2016-08-10T11:37:29+00:00","method":"GET","url":"\/master\/index.php\/settings\/admin\/sharing","user":"admin"}
{"reqId":"Ho2YP2oh2bf1gBTiY1Cx","remoteAddr":"::1","app":"PHP","message":"Undefined index: allowLinks at \/home\/schiesbn\/repos\/nextcloud\/server\/master\/settings\/templates\/admin\/sharing.php#45","level":3,"time":"2016-08-10T11:37:29+00:00","method":"GET","url":"\/master\/index.php\/settings\/admin\/sharing","user":"admin"}
{"reqId":"Ho2YP2oh2bf1gBTiY1Cx","remoteAddr":"::1","app":"PHP","message":"Undefined index: allowPublicUpload at \/home\/schiesbn\/repos\/nextcloud\/server\/master\/settings\/templates\/admin\/sharing.php#47","level":3,"time":"2016-08-10T11:37:29+00:00","method":"GET","url":"\/master\/index.php\/settings\/admin\/sharing","user":"admin"}
{"reqId":"Ho2YP2oh2bf1gBTiY1Cx","remoteAddr":"::1","app":"PHP","message":"Undefined index: enforceLinkPassword at \/home\/schiesbn\/repos\/nextcloud\/server\/master\/settings\/templates\/admin\/sharing.php#51","level":3,"time":"2016-08-10T11:37:29+00:00","method":"GET","url":"\/master\/index.php\/settings\/admin\/sharing","user":"admin"}
{"reqId":"Ho2YP2oh2bf1gBTiY1Cx","remoteAddr":"::1","app":"PHP","message":"Undefined index: allowPublicMailNotification at \/home\/schiesbn\/repos\/nextcloud\/server\/master\/settings\/templates\/admin\/sharing.php#59","level":3,"time":"2016-08-10T11:37:29+00:00","method":"GET","url":"\/master\/index.php\/settings\/admin\/sharing","user":"admin"}
{"reqId":"Ho2YP2oh2bf1gBTiY1Cx","remoteAddr":"::1","app":"PHP","message":"Undefined index: allowLinks at \/home\/schiesbn\/repos\/nextcloud\/server\/master\/settings\/templates\/admin\/sharing.php#63","level":3,"time":"2016-08-10T11:37:29+00:00","method":"GET","url":"\/master\/index.php\/settings\/admin\/sharing","user":"admin"}
{"reqId":"Ho2YP2oh2bf1gBTiY1Cx","remoteAddr":"::1","app":"PHP","message":"Undefined index: allowResharing at \/home\/schiesbn\/repos\/nextcloud\/server\/master\/settings\/templates\/admin\/sharing.php#74","level":3,"time":"2016-08-10T11:37:29+00:00","method":"GET","url":"\/master\/index.php\/settings\/admin\/sharing","user":"admin"}
{"reqId":"Ho2YP2oh2bf1gBTiY1Cx","remoteAddr":"::1","app":"PHP","message":"Undefined index: allowGroupSharing at \/home\/schiesbn\/repos\/nextcloud\/server\/master\/settings\/templates\/admin\/sharing.php#79","level":3,"time":"2016-08-10T11:37:29+00:00","method":"GET","url":"\/master\/index.php\/settings\/admin\/sharing","user":"admin"}
{"reqId":"Ho2YP2oh2bf1gBTiY1Cx","remoteAddr":"::1","app":"PHP","message":"Undefined index: onlyShareWithGroupMembers at \/home\/schiesbn\/repos\/nextcloud\/server\/master\/settings\/templates\/admin\/sharing.php#84","level":3,"time":"2016-08-10T11:37:29+00:00","method":"GET","url":"\/master\/index.php\/settings\/admin\/sharing","user":"admin"}
{"reqId":"Ho2YP2oh2bf1gBTiY1Cx","remoteAddr":"::1","app":"PHP","message":"Undefined index: allowMailNotification at \/home\/schiesbn\/repos\/nextcloud\/server\/master\/settings\/templates\/admin\/sharing.php#89","level":3,"time":"2016-08-10T11:37:29+00:00","method":"GET","url":"\/master\/index.php\/settings\/admin\/sharing","user":"admin"}
{"reqId":"Ho2YP2oh2bf1gBTiY1Cx","remoteAddr":"::1","app":"PHP","message":"Undefined index: allowShareDialogUserEnumeration at \/home\/schiesbn\/repos\/nextcloud\/server\/master\/settings\/templates\/admin\/sharing.php#104","level":3,"time":"2016-08-10T11:37:29+00:00","method":"GET","url":"\/master\/index.php\/settings\/admin\/sharing","user":"admin"}
{"reqId":"Ho2YP2oh2bf1gBTiY1Cx","remoteAddr":"::1","app":"PHP","message":"Undefined index: fileSharingSettings at \/home\/schiesbn\/repos\/nextcloud\/server\/master\/settings\/templates\/admin\/sharing.php#108","level":3,"time":"2016-08-10T11:37:29+00:00","method":"GET","url":"\/master\/index.php\/settings\/admin\/sharing","user":"admin"}
````

schiessle commented:
If I start with a complete fresh installation my app (ServerInfo) doesn't get added to the admin page. I first have to increase the version in the info.xml to run the update for the app at least once. New apps should be added directly.

schiessle commented:
If I change the priority, both in the IAdmin and the ISettings implementation and increase the version number in the info.xml the update will add the same app a second time to oc_admin_settings with the new priority. I would expect that it gets replaced.

In oc_admin_section nothing change, I still have only one entry with the previous priority. But maybe this is not needed anyway, see my question here https://github.com/nextcloud/server/pull/796#discussion_r74227738

schiessle commented:
Sorry for the many comments. But this was just the stuff I discovered while creating the admin page for the ServerInfo app. Beside that it was super easy and the API feels really good!

blizzz commented:
Currently working on switching from OCP\Template to TemplateResponse as return value form IAdmin, because it's nicer and already present in several apps. Changed user_ldap locally successful, now trying to adjust updatenotifications, where i somewhat struggle with autoloading yet.

blizzz committed:
more admin page splitup improvements

* bump version to ensure tables are created
* make updatenotification app use settings api
* change IAdmin::render() to getForm() and change return type from Template to TemplateResponse
* adjust User_LDAP accordingly, as well as built-in forms
* add IDateTimeFormatter to AppFramework/DependencyInjection/DIContainer.php. This is important so that \OC::$server->query() is able to resolve the
constructor parameters. We should ensure that all OCP/* stuff that is available from \OC::$server is available here. Kudos to @LukasReschke
* make sure apps that have settings info in their info.xml are loaded before triggering adding the settings setup method

LukasReschke commented:
> If I start with a complete fresh installation my app (ServerInfo) doesn't get added to the admin page. I first have to increase the version in the info.xml to run the update for the app at least once. New apps should be added directly.

Also could reproduce this locally. Migration was straight forward however: https://github.com/nextcloud/user_saml/pull/29

blizzz commented:
Yes, apps need to increase the version number. I don't see anything wrong with it.

LukasReschke commented:
> Yes, apps need to increase the version number. I don't see anything wrong with it.

It doesn't work on initial installation. Just checkout https://github.com/nextcloud/user_saml/pull/29 and then install Nextcloud and then enable the app.

blizzz commented:
> It doesn't work on initial installation. Just checkout nextcloud/user_saml#29 and then install Nextcloud and then enable the app.

Ok, misunderstood. I have an idea, going to fix it.

jancborchardt commented:
So it seems like this will still take quite some time to do properly, or is that not so? cc @blizzz @karlitschek cause I thought this would be for Nextcloud 10.

karlitschek commented:
We have too many open issues. Including this one. So I suggest to try to do the RC tomorrow instead of today.

blizzz commented:
> So it seems like this will still take quite some time to do properly, or is that not so? cc @blizzz @karlitschek cause I thought this would be for Nextcloud 10.

Yes, it needs more time. Nobody wanted a hacky solution, which would not allow for certain design goals anyway.

MorrisJobke commented:
> So it seems like this will still take quite some time to do properly, or is that not so? cc @blizzz @karlitschek cause I thought this would be for Nextcloud 10.

I would avoid to add this to 10. This will cause trouble.

karlitschek commented:
:-( We sort of need this with the latest changes. lets talk tomorrow

blizzz committed:
fix registration of admin settings and section on app install

blizzz committed:
fixes, improvements, and another app:

* setupSettings now also triggered on enable
* fixes detection of present admin section or settings in the DB
* add update routine in such cases
* encryption app migrated

blizzz commented:
Considering to rename IAdmin to ISettings, because in future this might be used for/to integrate Personal settings and on that level there is no differentiation between Admin and Personal. This happens in info.xml.

> :-( We sort of need this with the latest changes. lets talk tomorrow

Yeah, we should. It's more work than initially anticipated.

blizzz committed:
rename IAdmin to ISettings, the interface is not bound to a specific settings scope

blizzz committed:
adjust federatedfilesharing

schiessle commented:
>> It doesn't work on initial installation. Just checkout nextcloud/user_saml#29 and then install Nextcloud and then enable the app.

> Ok, misunderstood. I have an idea, going to fix it.

@blizzz looking at the commits it sounds like this should have been fixed with https://github.com/nextcloud/server/pull/796/commits/0fc34c99f4e1dbe26e5074102f48f75d70fe97d0 But it still seems to be broken with apps which are enabled by default.

steps to test:

- checkout your branch
- checkout https://github.com/nextcloud/serverinfo/pull/18 to the apps folder
- install Nextcloud
-> ServerInfo is not the the admin settings. If I increase the version number and let the update run it appears in the settings.

blizzz committed:
adjust files_external

blizzz commented:
@schiessle thx for the reproduction steps, I'll investigate it

blizzz commented:
Seriously, we need to clean up lib/private/Installer.php and lib/private/legacy/app.php at some point, it is utterly confusing and chaotic.

blizzz committed:
make sure shipped apps also setup their admin settings on a fresh install

blizzz commented:
@schiessle fixed!

blizzz committed:
superfluous

blizzz committed:
adjust Theming app

blizzz committed:
adjust files app, and integrate files_external into that section

blizzz committed:
superfluous

blizzz committed:
adjust federation app

blizzz committed:
adjust systemtags app

blizzz commented:
Another milestone… all apps within this repo are adjusted now. For systemtags I enabled a build-in category "Collaboration", however there is nothing else inside yet. No idea where else to put it.  Maybe it becomes more useful when looking at other apps outside of this repo.

LukasReschke committed:
Add missing since annotations

LukasReschke committed:
Add since tags to class

LukasReschke committed:
Merge branch 'master' into implement_712

LukasReschke committed:
Resolve conflict

LukasReschke commented on apps/systemtags/appinfo/app.php:
```diff
@@ -78,9 +78,6 @@ function() {
-
 $l = \OC::$server->getL10N('systemtags');
 
 \OCA\Files\App::getNavigationManager()->add(
```
Mhm?

PHP Fatal error:  Class 'OCA\Files\App' not found in /drone/src/github.com/nextcloud/server/apps/systemtags/appinfo/app.php on line 83

blizzz replied:
:confused: Unfortunate app load order? Runs locally beyond this point, and succeeds.

blizzz replied:
Also, the Tags are shown in the sidebar and the GUI is working, so on a web server run it works, too.

blizzz replied:
Turns out, it is different behaviour on drone compared to my machine. Systemtags is installed as a shipped app before files. Because we need to instantiate the ISettings implementation, the app is loaded after install. In case of our CI the error is thrown, however not on mine system.

This can be fixed by not loading the app, instead of only adding registering the autoloader. app.php will not be executed in that case. In case  of the shipped app installer this is even done before. Relying on the autoloader means that you cannot use foreign classes in the ISettings implementor constructor. But, for one depending on non-OCP namespaced things is discouraged, kind of forbidden, besides that I have doubts this would work at all anyway due to the way how we resolve the constructor parameters.

Going a step back, ideally an instantiation of the ISettings implementor would not be necessary, however the class carries information we like to have in the DB. Since we basically need properties, we could work with static methods. Typically those return just the section identifier and an ordering number. Thus, testing is not a problem, static can be tolerable while still being required by the interface.

Alternatively, we move these information also into the info.xml file, extending the <admin> section with sub-elements ``class``, ``section`` and ``priority``. However, we'll need to detect empty sections at some point, so moving the section identifier out of it is not helpful, and neither is duplicating the value. 

My conclusion: autoloading should be fine. In fact, this is also already utilized in the shipped apps installer.

blizzz replied:
With switching to autoloading I have an issue with autoloading on Theming left… Need to understand this yet.

blizzz replied:
Unrelated. Actually with loadApp it is even worse, some more errors thrown on install, so this change is nice. Push this and look further into theming.

blizzz committed:
register app autoload instead of loading apps

blizzz commented:
@schiessle maybe you have an idea. On a fresh install, setting up the Admin settings for the Theming app fails, because \OCA\Theming\Template cannot be resolved. For debugging reasons I removed it's constructor to have no further dependencies. It did not help. Setting it up worked (as in was written to the DB) only, if I removed Template from the Admin constructor. https://github.com/nextcloud/server/pull/796/files#diff-c8a65918a2a85ee57b57bc8b2dab2e34R47

However, FederatedFileShare uses a similar construct, resolving FederatedShareProvider works: https://github.com/nextcloud/server/pull/796/files#diff-20d6741bf94cd5e1bf6be579a16641e8R35

Debugging the resolver does not lead me anywhere, yet alone the thrown Exception, which is misleading.

schiessle committed:
rename "Tenmplate" to "ThemingDefaults" to make the auto loader happy"

blizzz committed:
rename remaining occurences of OCA/Theming/Template

blizzz committed:
added some missed diagnosis output

blizzz committed:
fix theming tests

blizzz commented:
@jancborchardt your statement on arrangement of sections, their names, and their settings?

blizzz committed:
mark current section

jancborchardt commented:
@blizzz 

- [ ] casing in the navigation is wrong. Only capitalize the first word
- [ ] »Additional settings« is empty
- [ ] let’s put »File handling« and »Email server« into »Additional settings« as they feel out of place in their respective categories
- [ ] »Updater« doesn’t need a heading and should be directly part of the »Version« section. It belongs together.
- [ ] »Encryption« and »Collaboration« feel a bit lost on their pages. Not sure if they should be moved to »Additional settings« though.
- [ ] While we’re at it, can we merge this with the »Personal« settings to one »Settings« page? Then we’ll have true simplification.

blizzz commented:
@jancborchardt "Additional settings" is a fallback for apps who did not adjust to the new mechanism yet. Agree, it should be hidden if empty.

jancborchardt commented:
@blizzz I know, but »File handling« and »Email server« are so unimportant and out of place in the other settings that they should be in »Additional«

blizzz commented:
>  While we’re at it, can we merge this with the »Personal« settings to one »Settings« page? Then we’ll have true simplification.

In general absolutely, the mechanism is designed to make this possible. However, certainly not for Nc 10.

blizzz commented:
> »Encryption« and »Collaboration« feel a bit lost on their pages. Not sure if they should be moved to »Additional settings« though.

Encryption has a tad more on the page when the corresponding app is enabled.

I am open on Collaboration indeed.

However stuffing things into Additional Settings and having a couple of other apps enabled will make that page long again.

jancborchardt commented:
@blizzz then I guess Collaborative tags most likely correspond to »Sharing«?

blizzz committed:
change casing in section display names

blizzz committed:
make updatenotification settings appear as if it is part of the version part in server settings

blizzz commented:
> »Updater« doesn’t need a heading and should be directly part of the »Version« section. It belongs together.

FYI, updater is a separate app. I removed the heading and lowered the priority so that part follows directly. Nevertheless, there is some whitespace as usual. Or would be. I added another CSS class ``followupsection`` which basically is the same as section, but has the top-padding set to 0 and the top-margin to -30px so it appears as if it would be one part. If this is stupid, please fix it, I am not a CSS hero.

blizzz committed:
move systemstags to sharign section, drop collaboration section

blizzz committed:
change federatedfilesharing css class to followupsection to reduce whitespace

blizzz committed:
move mail settings and file handling to additional ones, thus files_external gets its own section

blizzz committed:
simplify encryption manager fetching in DIContainer

blizzz commented:
> @blizzz then I guess Collaborative tags most likely correspond to »Sharing«?

Sounds reasonable, done.

@jancborchardt I understood I leave Encryption in place, correct?

@schiessle fyi, I saw a similar issue now with files_external and its GlobalStoragesService dependency. The others it requests are good, but this one fails. It's a bit different though, because it is available from Server.php which uses \OC_Mount_Config to fetch it which is part of files_external. However, this is not called, if I add it to the DIContainer it will be but dies with an unknown \OC_Mount_Config …

LukasReschke commented:
As discussed with @blizzz I'll write the unit tests for this.

blizzz committed:
satisfy dependencies for files_external

blizzz commented:
> @schiessle fyi, I saw a similar issue now with files_external and its GlobalStoragesService dependency. The others it requests are good, but this one fails. It's a bit different though, because it is available from Server.php which uses \OC_Mount_Config to fetch it which is part of files_external. However, this is not called, if I add it to the DIContainer it will be but dies with an unknown \OC_Mount_Config …

Missing nested dependency …

nickvergessen commented:
Btw tables are still "admin*" should we adjust this as well before merge, to be prepared for personal settings?

blizzz commented:
> Btw tables are still "admin*" should we adjust this as well before merge, to be prepared for personal settings?

we can change this now  and add another col or add another pair of tables later. I don't have a strong opinion on this. I would however prefer to stabilize it to get the RC out asap, although the impact would not be big on this one.

LukasReschke committed:
Add unit tests

LukasReschke committed:
Use MockBuilder instead of createMock

CI uses an older PHPUnit

LukasReschke commented:
Tests added.

blizzz committed:
attempt to remove section and settings entries when an app got disabled

LukasReschke committed:
Merge branch 'master' into implement_712

blizzz committed:
check registered sections and settings after an app got updated to garbage collect orphaned classes

blizzz committed:
adopt to Controller constructor changes

blizzz commented:
> don't do a full page reload when switching sections

Unfortunately this does not seem to be achievable in short term. The problem is not the loading itself. This works just fine. The problem however is loading of additional (and repeatedly!) javascript resources. The initialization needs to be redone, which is happening with some of our scripts. And the scripts are not robust against being loaded multiple times (i.e. when opening Theming, then Server Info, and Theming again) and just break.

blizzz committed:
final db indexes

blizzz commented:
Request for final previews @nickvergessen @schiessle @jancborchardt

blizzz commented:
Btw, for those who already have the tables in the database and attempt an upgrade with the latest commit, it will fail (on mysql), because a primary key was missing. This taken care of with the re-evaluation of indexes and won't happen in production.

jospoortvliet commented:
last comment shakes my confidence in @icewind1991 as he asserts that running master is as easy if not easier than running releases ;-)

/me hides again

LukasReschke commented:
> last comment shakes my confidence in @icewind1991 as he asserts that running master is as easy if not easier than running releases ;-)

Well. Running master is, this is *NOT* master. This is a different branch 😉

jospoortvliet commented:
> you're yourself to blame ;-)
I generally blame you, but fair enough.

schiessle commented:
Just ported the usage report app (https://github.com/nextcloud/survey_client/pull/13) everything works great, so :+1:

schiessle commented:
Just saw that this still happens:

- [ ] https://github.com/nextcloud/survey_client/pull/13

schiessle commented:
All the stuff removed here https://github.com/nextcloud/server/pull/796/files#diff-7ce542aade9c380d3b2e16162d559256L1 need to be added here probably https://github.com/nextcloud/server/pull/796/files#diff-9ee3df1a8b757f23c30063a393a3d551R52

This was done for the mail settings but a lot of other settings, including the sharing settings are mission... @blizzz am I right?

blizzz commented:
> This was done for the mail settings but a lot of other settings, including the sharing settings are mission... @blizzz am I right?

No, they are split up. Some parts are in Server, some are in Additional Setting. And Sharing settings are in Sharing.

blizzz commented:
> Just saw that this still happens:

> - [ ] https://github.com/nextcloud/survey_client/pull/13

@schiessle wrong link?

schiessle commented:
> No, they are split up. Some parts are in Server, some are in Additional Setting. And Sharing settings are in Sharing.

yes, I was to fast and edited my comment right before you commented... Still it seems like we removed here setting parameters from the general admin page which we added in no other place. At least I discovered it for sharing and just at the moment for encryption: https://github.com/nextcloud/server/pull/796/files#diff-7ce542aade9c380d3b2e16162d559256L132 So the encryption settings no longer show the encryption modules.

schiessle commented:
@blizzz why do you still see the old link. Edited the post directly after I created it... This is the correct link: https://github.com/nextcloud/server/pull/796#issuecomment-238841484

blizzz commented:
> @blizzz why do you still see the old link. Edited the post directly after I created it... This is the correct link: #796 (comment)

Glitch in the matrix?

blizzz commented:
@schiessle thx for re-checking, will fix it.

blizzz committed:
fix missing parameters in sharing settings page

blizzz commented:
@schiessle done

blizzz committed:
add missing encryption modules to settings

LukasReschke commented:
Conflicts :)

blizzz committed:
for new we cannot have nested settings, default module is only appended if available

blizzz committed:
Merge branch 'master' into implement_712

blizzz commented:
> Conflicts :)

:v:

blizzz committed:
adjust tests to latest changes

schiessle commented:
looks good now :+1:

LukasReschke commented:
LGTM

LukasReschke commented:
Oh. Backporting this is fun 😨  – So many conflicts…

LukasReschke commented:
Backport attempt at https://github.com/nextcloud/server/pull/894, let's see…
